src/Microsoft.ServiceHub.Framework/StreamExtensions.cs

```diff
@@ -2,13 +2,15 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Diagnostics.CodeAnalysis;
+using System.IO.Pipes;
 using Microsoft.Win32.SafeHandles;
 
 namespace Microsoft.ServiceHub.Framework;
 
 /// <summary>
 /// A class containing extension methods for <see cref="Stream"/>.
 /// </summary>
+[Obsolete("This API may soon be moved.")]
```

Comment:
Can we delete this method?

It *does* provide access to the SafePipeHandle on the `ServiceHubPipeStream`, which is an internal class. Does that need to be exposed externally?

Reply:
No, we need this method for now unless we also make ServiceHubPipeStream public. We use the SafePipeHandle in the VS repo to get the PID of the process that it belongs to. We then compare that PID to the set of Host processes that the Controller has spawned to make sure its actually a Host process.

We can get rid of this method if we had a different way of getting either the SafePipeHandle of the Stream or the PID of the process that it belongs to.

Reply:
How about we remove the 'extensions method' aspect to it then, so at least we're not polluting the completion lists of all Stream users when it's only used in ServiceHub?
And what if I moved it to another class, like ServerFactory?

Reply:
I'm fine with both of those changes